pkg/controller/appdefinition/pullappimage.go

```diff
@@ -11,9 +12,25 @@ import (
 	"github.com/acorn-io/acorn/pkg/tags"
 	"github.com/acorn-io/baaah/pkg/router"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func (i *imagePuller) remoteImageDigest(ctx context.Context, c client.Reader, namespace, image string) (string, error) {
```

Comment:
sorry, i meant these functions for the struct as well. would move them all down to the bottom